### ISSUE TITLE
Question about the evaluation method

### ISSUE THREAD

**Issue by JustinYuu** [FloretCat/CMRAN#1]
Hi!
Thanks a lot for sharing the code generously, especially when there are few available codes in this task! I believe your codes will help many researchers in this field, including me.
However, I have a small question about your evaluation method. It seems that the validation set is not used in the training process. Instead, the test set is directly used during every training epoch and the best performance in the test set is chosen as the reported performance.  I wonder if this training strategy could lead to the overfitting on the test set, and what's the difference between the test phase and the test progress in the training phase by doing so?
Please remind me if I misunderstand the training process in the released codes, and thanks again for your generous sharing!

**Comment by FloretCat** [comment]
Thanks for your interest in our work!

（1）Since the test set is publicly available, we did not include the code for the evaluation on the validation set in our released code for convenience. The validation set will play a greater role when the test set is withheld or the time cost for the evaluation is large. To exactly explore how well our method performs on the test set, it is a common behavior that evaluating the model on the test set every training epoch especially when the time cost for evaluation is not very large. This is a trade-off between the time cost and the precision of the evaluation of a method.
（2）No difference. But if you need to evaluate a model that you have not evaluated on the test set, the script (i.e, supv/weak_test.sh) for only running testing may be useful.

**Comment by JustinYuu** [comment]
Thanks for your timely response.
Since the validation set is also publicly available and it has the same size as the test set(402 videos), I think the time cost of using the validation and test set during the training phase is the same.  However, just picking the model that has the best performance on the test set may lead to overfitting, thus it's hard to judge whether the selected model is only valid for the test set or for any data distribution. I think the standard procedure is to choose the best model on the validation set during training and test the selected model on the test set so as to ensure the model performance is stable on any data distribution, i.e., the test set should only be used once, just like the way [Tian et al.](https://github.com/YapengTian/AVE-ECCV18/blob/master/supervised_main.py) did.

**Comment by FloretCat** [comment]
> Thanks for your timely response.
> Since the validation set is also publicly available and it has the same size as the test set(402 videos), I think the time cost of using the validation and test set during the training phase is the same. However, just picking the model that has the best performance on the test set may lead to overfitting, thus it's hard to judge whether the selected model is only valid for the test set or for any data distribution. I think the standard procedure is to choose the best model on the validation set during training and test the selected model on the test set so as to ensure the model performance is stable on any data distribution, i.e., the test set should only be used once, just like the way [Tian et al.](https://github.com/YapengTian/AVE-ECCV18/blob/master/supervised_main.py) did.

Yes, you are exactly right. The current released version is just used to explore the exact performance of the method on the test set, without considering other data. We will update this later.

Thanks！
